blivetgui/dialogs/message_dialogs.py

```diff
@@ -179,6 +180,63 @@ def run(self):
         return response == Gtk.ResponseType.OK
 
 
+class ConfirmDeleteDialog(object):
+    """ General confirmation dialog
+    """
+
+    def __init__(self, parent_window, device, parents=None):
+
+        builder = Gtk.Builder()
+        builder.set_translation_domain("blivet-gui")
+        builder.add_from_file(locate_ui_file('confirm_delete_dialog.ui'))
+        self.dialog = builder.get_object("confirm_delete_dialog")
+
+        self.dialog.set_transient_for(parent_window)
+
+        title = _("Confirm delete operation")
+        msg = _("Are you sure you want delete device {name}?").format(name=device.name)
+
+        self.dialog.set_markup("<b>" + title + "</b>")
+        self.dialog.format_secondary_text(msg)
+
+        self.parent_check = builder.get_object("parent_check")
+        self.parent_label = builder.get_object("parent_label")
+
+        if parents:
+            check_text = _("Also delete following parent devices of {name}:").format(name=device.name)
+            self.parent_check.set_label(check_text)
+            self.parent_check.set_active(True)
+
+            parent_text = ""
+            for parent in parents:
+                # TRANSLATORS: device specification, e.g. "100 MiB partition /dev/sda1"
+                parent_text += _(" • {size} {type} {path}\n").format(size=parent.size,
```

Comment:
Good point. I had a longer text there and didn't delete the comment (and `_`) after the change.